datafusion/src/physical_plan/parquet.rs

```diff
@@ -582,14 +389,244 @@ impl ParquetExec {
+                                    }
+                                }
+                            }
+                            ParquetStatistics::Double(s) => {
+                                if let DataType::Float64 = fields[i].data_type() {
+                                    if s.has_min_max_set() {
+                                        if let Some(max_value) = &mut max_values[i] {
+                                            match max_value.update(&[
+                                                ScalarValue::Float64(Some(*s.max())),
+                                            ]) {
+                                                Ok(_) => {}
+                                                Err(_) => {
+                                                    max_values[i] = None;
+                                                }
+                                            }
+                                        }
+                                        if let Some(min_value) = &mut min_values[i] {
+                                            match min_value.update(&[
+                                                ScalarValue::Float64(Some(*s.min())),
+                                            ]) {
+                                                Ok(_) => {}
+                                                Err(_) => {
+                                                    min_values[i] = None;
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                            _ => {}
+                        }
+                    }
+                }
+
+                if limit.map(|x| num_rows >= x as i64).unwrap_or(false) {
+                    break;
+                }
+            }
+        }
+        let column_stats = (0..num_fields)
+            .map(|i| {
+                let max_value = match &max_values[i] {
+                    Some(max_value) => max_value.evaluate().ok(),
+                    None => None,
+                };
+                let min_value = match &min_values[i] {
+                    Some(min_value) => min_value.evaluate().ok(),
+                    None => None,
+                };
+                ColumnStatistics {
+                    null_count: Some(null_counts[i] as usize),
+                    max_value,
+                    min_value,
+                    distinct_count: None,
+                }
+            })
+            .collect();
+
+        let statistics = Statistics {
+            num_rows: Some(num_rows as usize),
+            total_byte_size: Some(total_byte_size as usize),
+            column_statistics: Some(column_stats),
+        };
+        // remove files that are not needed in case of limit
+        let mut filenames = filenames;
+        filenames.truncate(total_files);
```

Comment:
do you also need to truncate `schemas` as well?

Reply:
looks like it.